bin/test-create-block.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# This script validates whether `npm init @wordpress/block` works properly
+# with the latest changes applied to the `master` branch. It purpousfuly
+# avoids installing `@wordpress/scripts` package from npm when scaffolding
+# a test block and uses the local package by executing everything from the
+# root of the project.
+
+# Exit if any command fails.
+set -e
+
+status () {
+	echo -e "\n\033[1;34m$1\033[0m\n"
+}
+
+status "Scaffolding block..."
+rm -rf esnext-test
```

Comment:
I'd suggest to use a temporary directory for this via `TMPDIR=$(mktemp -d)` and a trap command to clean it up via `trap 'rm -rf "$TMPDIR"' EXIT`

Reply:
How would I run all commands in the temp folder? I guess it's possible but I simply don't know :)

Reply:
Maybe we should add an option for the output directory? 😄  `npx wp-create-block esnext-test --output "$TMPDIR"`

Reply:
It sounds like a nice option to add in another PR. For this PR, could we store references to `wp-create-block` and `wp-scripts` under variable and use them later when executing commands inside the temp dir?

Reply:
It looks like shell script isn't my thing. I don't even know what to start with without googling. I'm happy to improve it while updating to Node.js in a follow-up PR unless someone else feels brave enough to add changes to this branch 🙂

Reply:
I added the cleanup task in https://github.com/WordPress/gutenberg/pull/23195/commits/041ab1d14ffe052c127b56682862c524afa2eb43. I think this is good to 🚢  now.

Reply:
Thank you very much ❤️